src/ads/native-ad/NativeAdView.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import React, { useRef } from 'react';
+import { ViewProps } from 'react-native';
+import { NativeAd } from './NativeAd';
+import { NativeAdContext } from './NativeAdContext';
+import GoogleMobileAdsNativeView from '../../specs/components/GoogleMobileAdsNativeViewNativeComponent';
+
+export type NativeAdViewProps = ViewProps & {
+  nativeAd: NativeAd;
+};
+
+export const NativeAdView = (props: NativeAdViewProps) => {
+  const { nativeAd, children, ...viewProps } = props;
+  const ref = useRef<React.ElementRef<typeof GoogleMobileAdsNativeView>>(null);
```

Comment:
We could add an effect here to destroy the nativeAd on "unmount" so the user doesn't have to do this himself?

Reply:
I think destroying ad is at app developer's discretion, because loading ad was initiated by app developer. As ad was created outside of the NativeAdView, NativeAdView doesn't have permission to destroy it, for my opinion.
Also maybe there's very specific case that want to unmount ad view and display it somewhere in other view with same ad?

Reply:
I agree on your opinion.